File: backend/src/main/java/com/springboot/backend/luismartinez/billingsapp/billingbackend/entities/Customer.java
package com.springboot.backend.luismartinez.billingsapp.billingbackend.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Entity
@Table(name = "customers")
public class Customer {

    // Getters and Setters
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotEmpty(message = "First name cannot be empty")
    private String firstName;

    @NotEmpty(message = "Last name cannot be empty")
    private String lastName;

    @NotEmpty(message = "Email cannot be empty")
    @Email(message = "Invalid email format")
    @Column(unique = true)
    private String email;

    @NotEmpty(message = "Address cannot be empty")
    private String address;

    @NotEmpty(message = "Phone cannot be empty")
    private String phone;

}

File: frontend/lminvoice/src/app/features/customers/models/customer.ts
export interface Customer {
  id?: number;
  firstName?: string;
  lastName?: string;
  name: string;
  email: string;
  address?: string;
  phone?: string;
}

export interface CustomerPayload {
  firstName: string;
  lastName: string;
  email: string;
  address: string;
  phone: string;
}
